libs/ast_generic/AST_generic.ml

```diff
@@ -303,7 +303,16 @@ and resolved_name_kind =
    * resolved_name, but we also want to match the pattern `foo.bar` so we will
    * store ['foo', 'bar'] as an alternate name.
    * *)
-  | ResolvedName of dotted_ident * dotted_ident list
+  | ResolvedName of unique_name * unique_name list
+
+(* This is not relevant for matching purposes.
+   `unique_name` is a record of all the information necessary to
+   disambiguate two `ResolvedName`s from each other.
+   In particular, two `ResolvedName`s may indicate two overloaded functions,
+   which are named the same. Then, we need to use the function signature information
+   to disambiguate them from each other.
```

Comment:
I still don't understand. sid is supposed to be the thing we use to disambiguate two names (for example
in a function you can have multiple times the same name 'a' for a variable declared in nested blocks, but it does
not matter because we use a different sid for each of those. So why not use sid here for the same purpose, to make sure 2 references to 2 different overloaded function have a different sid ...

Reply:
In fact, where do you ever use this tok in the matching code?

Reply:
@aryx I believe that we came to the idea to use the `tok` for disambiguation together. Here's what Nat said on the other PR about this:

> Semgrep uses sids but I believe they are created during name resolution. We need to be able to construct Name.ts ahead of name resolution as part of creating the SIGs. I'm sure we could still come up with a scheme to use sids for this, and I think it would be valuable work, but I don't think it would be more difficult to do after merging this PR, and I don't see a reason to increase the complexity of this change beyond what is necessary.

I believe that if we wanted to use `sid`s, it would not be very hard to make a change to do so after the fact, but right now it would add a fair amount of complexity to the PR, which is already quite long. I have no real objections to it, but I would strongly prefer that we hold off until after the initial change, so it can be considered and evaluated separately.

Reply:
Because if we use sids like they're supposed to be used, we don't need https://github.com/returntocorp/semgrep/pull/6788 at all